### ISSUE TITLE
Unrecognised keys in argument objects can lead to invalid CQL queries

### ISSUE THREAD

**Issue by szmoore** [Flux159/cassie-odm#8]
This is an example of an error message generated by 0.2.2 using cassandra-driver 3.2.0 with console debugging of intermediate variables

```
debug argStringArray = ""
debug argument string:  WHERE  fishUid =?  AND  AND  day =?  AND  fishType =?
debug argument params: 4c93b46f-2e61-4174-9b18-f93954cca4e3,1,Barramundi
2017-03-30T07:08:04.904Z ERROR Could not get data: ResponseError: line 1:45 no viable alternative at input 'AND' (... 
WHERE  fishUid =?  AND  [AND]...)
```

The double `AND` occured because of an incorrect use of `$gte` and `$lte` where the driver only supports `$gt` and `$lt` - in [this for loop](https://github.com/Flux159/cassie-odm/blob/master/lib/model.js#L80)

If the value of `vkey` is none of the explicitely checked values in the if statements, then the `AND` can be included without `tempArgStringArray` and `tempArgValueArray` being populated.

One way to reproduce follow example for `$gt` but substitute `$gt` for `$gte`.

The input is technically an error, but the message that results is confusing; it would be nice if the invalid arguments were checked and an appropriate error message raised instead of falling back to a failure in `cassandra-driver`.

**Comment by Flux159** [comment]
Hey @szmoore, it seems like CQL supports '>=', '<=', '<', and '>' syntax (at least according to http://cassandra.apache.org/doc/latest/cql/dml.html#select), but cassie is parsing all $gt and $lt as '>=' and '<=' - this is definitely a logic error in cassie.

I don't think that ignoring '$gte' and '$lte' is a good idea since technically they should be supported by CQL.

It would probably be better to change $gt and $lt to actually produce '<' and '>', and add $gte and $lte capability while incrementing the version to 0.3.0 and marking this as a breaking change in the API and documenting the CQL querying options better in the Readme.md.

I think I can make these changes fairly easily, but will need to write the appropriate tests to ensure correctness. I'll reply back as soon as I can about this.

Thanks for the bug report.

**Comment by Flux159** [comment]
This should be fixed in 0.3.0 - Let me know if that works for you. I added a test in /readme-tests/gtquery_test.js and added to the Readme about $gt, $lt, $gte, and $lte query syntax in https://github.com/Flux159/cassie-odm#crud-create-read-update-delete-operations under "Read Example 3 (Querying using $gt, $lt, $gte, $lte, or $in):".

Also added Release notes about breaking changes between 0.1.x, 0.2.x, and 0.3.x.

I'll leave the issue open for a few days and close it if there's no response or if it works for you.

Thanks.

**Comment by Flux159** [comment]
Released 0.3.1 as well (no code changes from 0.3.0, but had to cleanup the README and other docs and wanted to make sure that npm reflected those changes).

**Comment by szmoore** [comment]
v0.3.1 fixes the issue for me, which I was working around by replacing $gte/$lte with $gt/$lt. 
Thanks @Flux159
